ports/esp32s2/supervisor/port.c

```diff
@@ -118,11 +118,12 @@ uint32_t port_get_saved_word(void) {
 uint64_t port_get_raw_ticks(uint8_t* subticks) {
     struct timeval tv_now;
     gettimeofday(&tv_now, NULL);
-    uint64_t all_subticks = (uint64_t)tv_now.tv_usec / 32768;
+    // convert usec back to ticks
+    uint64_t all_subticks = (uint64_t)tv_now.tv_usec / 1024;
```

Comment:
I don't think this is correct because all_subticks should range from 0 to 32767 which would be achieved by dividing 1000000 by 30.517. We can approximate this with the change below. Line 125 should be preserved.

```suggestion
    uint64_t all_subticks = (uint64_t)(tv_now.tv_usec * 2) / 71;
```

Reply:
With tv_usec having a value between 1 and 1 million, multiplying it by 2 and dividing by 71 give us values from 1 to 28168. Since a tick is 1/1024th of a second,  that could add to the return value by up to 27 seconds. It seems correct to me to multiply seconds by 1024, and add subticks of 1000000 divided by 1024, which is what the original fix did. That would give us subticks of 1 to 1000, which when added to seconds*1024 would be the right number of ticks. 
Testing of the proposed fix showed times much greater than the call to sleep() requested.

Reply:
How are you measuring the sleep duration? I think there is more going on with FreeRTOS that makes it harder to measure.

 The `all_subticks` should be divided by 32 before being added to `seconds * 1024` because one tick is 32 subticks. (Most boards have 32.768khz crystals which is 1024 * 32)